Automattic/wp-calypso
# Themes: Fix back button on sheet

<img width="278" alt="screen shot 2016-05-18 at 12 25 40" src="https://cloud.githubusercontent.com/assets/7767559/15356962/dc040b1c-1cf3-11e6-8f5b-87a8ad757ac5.png">

Back button will:
* Always go to /design...
* Maintain any filters or search args that were previously in place
* go to /design/ if arrived cold at /theme/...

I tried the approach of having the back path as a URL query arg, but it felt fragile. This approach stores the path in the global app state.

**To Test**
Try using the _< BACK_ button in the top left of the theme sheet in the following scenarios. Expected results as above.
* start from http://calypso.localhost:3000/design and choose details for a theme
* click around the different sections of the sheet
* activate a theme
* add a search at http://calypso.localhost:3000/design
* Use a filter (free/premium/all) at http://calypso.localhost:3000/design
* come in cold to /theme/{theme}
* logged-in
* logged-out
